### PR TITLE
update recommended extensions and settings

### PR DESCRIPTION
**Pull request by xavdid-stripe**
Removing old extensions, adding recommended settings

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,12 +4,11 @@
...
+
   // Formatting
   "editor.formatOnSave": true,
-  "phpformatter.composer": true,
-  "phpformatter.arguments": [""],
-
-  // Linting
-  "phpcs.enable": true,
-  "phpcs.standard": "PSR2"
+  "intelephense.format.braces": "psr12"
```
Comment (xavdid-stripe) [inline .vscode/settings.json:13]: I think this is the same, but i'm not positive. I did see some minor bracket motion when saving a file, so i'm not sure.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {
   "recommendations": [
-    "felixfbecker.php-intellisense",
-    "ikappas.phpcs",
```
Comment (xavdid-stripe) [inline .vscode/extensions.json:4]: this is deprecated, so I pulled it

File: .vscode/extensions.json [hunk 3]
```diff
@@ -1,9 +1,8 @@
 {
   "recommendations": [
-    "felixfbecker.php-intellisense",
```
Comment (xavdid-stripe) [inline .vscode/extensions.json:3]: this has been removed from the marketplace

File: .vscode/extensions.json [hunk 4]
```diff
@@ -1,9 +1,8 @@
...
-    "ikappas.phpcs",
-    "emallin.phpunit",
-    "EditorConfig.editorconfig",
-    "sophisticode.php-formatter"
```
Comment (xavdid-stripe) [inline .vscode/extensions.json:7]: also deprecated

File: phpstan.neon.dist [hunk 5]
```diff
@@ -11,3 +11,7 @@ parameters:
 		- '#Unsafe usage of new static\(\).#'
 
 	reportUnmatchedIgnoredErrors: false
+
+	paths:
+		- lib
```
Comment (xavdid-stripe) [inline phpstan.neon.dist:16]: needed for the phpstan extension

**Review by xavdid-stripe — COMMENTED** [review]
Ok, please double check this one - it's the ecosystem I know least about. Feel free to push back!

File: .vscode/settings.json (hunk 1 above)
Comment (prathmesh-stripe) [reply .vscode/settings.json:13]: Can you tell me the files that were changed? I am wondering how this change would affect the formatter in our CI. I doubt they support PSR12.

Comment (xavdid-stripe) [reply .vscode/settings.json:13]: This PR didn't change any files directly - stuff would get changed as vscode users format/save things. 

That said, this isn't a great solution. It doesn't seem like a 1:1 replacement and not everyone on the team uses vscode. I wonder if we don't use vscode for formatting at all and lean on `make format` instead, which will be consistent across editors and CI.

In JS and other languages, the formatter and the vscode extension use the same program under the hood, so they always agree. I'm not sure PHP can have the same setup.

Comment (prathmesh-stripe) [reply .vscode/settings.json:13]: Let's see if this would break anything in the future. We can anyways fallback on the `make fmt` for formatting.

**Review by prathmesh-stripe — APPROVED** [review]

**Pull request merged**